# Block copying crap out of header files in the name of "portability" is
# never a good idea.  It breaks stuff.  In this case, it won't build under
# Linux 3.0.


--- busybox/include/libbb.h	2011-03-12 19:45:40.000000000 -0600
+++ busybox.new/include/libbb.h	2011-07-23 18:24:06.563634320 -0500
@@ -31,6 +31,7 @@
 #include <sys/mman.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
+#include <sys/sysinfo.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/wait.h>
@@ -100,25 +101,6 @@
 #if !defined __FreeBSD__
 char *dirname(char *path);
 #endif
-/* Include our own copy of struct sysinfo to avoid binary compatibility
- * problems with Linux 2.4, which changed things.  Grumble, grumble. */
-struct sysinfo {
-	long uptime;			/* Seconds since boot */
-	unsigned long loads[3];		/* 1, 5, and 15 minute load averages */
-	unsigned long totalram;		/* Total usable main memory size */
-	unsigned long freeram;		/* Available memory size */
-	unsigned long sharedram;	/* Amount of shared memory */
-	unsigned long bufferram;	/* Memory used by buffers */
-	unsigned long totalswap;	/* Total swap space size */
-	unsigned long freeswap;		/* swap space still available */
-	unsigned short procs;		/* Number of current processes */
-	unsigned short pad;			/* Padding needed for m68k */
-	unsigned long totalhigh;	/* Total high memory size */
-	unsigned long freehigh;		/* Available high memory size */
-	unsigned int mem_unit;		/* Memory unit size in bytes */
-	char _f[20 - 2 * sizeof(long) - sizeof(int)]; /* Padding: libc5 uses this.. */
-};
-int sysinfo(struct sysinfo* info);
 #ifndef PATH_MAX
 # define PATH_MAX 256
 #endif
diff -ru busybox/networking/Config.src busybox.new/networking/Config.src
--- busybox/networking/Config.src	2011-03-12 19:45:40.000000000 -0600
+++ busybox.new/networking/Config.src	2011-07-23 19:12:14.379897634 -0500
@@ -740,7 +740,7 @@
 
 config TCPSVD
 	bool "tcpsvd"
-	default y
+	default n
 	help
 	  tcpsvd listens on a TCP port and runs a program for each new
 	  connection.
diff -ru busybox/networking/Config.src busybox.new/networking/Config.src
--- busybox/networking/Config.src	2011-07-23 19:30:19.800800291 -0500
+++ busybox.new/networking/Config.src	2011-07-23 19:39:50.129877165 -0500
@@ -972,7 +972,7 @@
 
 config UDPSVD
 	bool "udpsvd"
-	default y
+	default n
 	help
 	  udpsvd listens on an UDP port and runs a program for each new
 	  connection.
